core/lpms.go

```diff
@@ -159,10 +177,23 @@ func (l *LPMS) SegmentRTMPToHLS(ctx context.Context, rs stream.RTMPVideoStream,
 					glog.Errorf("HLS Stream is nil")
 					return segmenter.ErrSegmenter
 				}
-				seg, err = s.PollSegment(segCtx)
+
+				for i:=0; i < RetryCount; i++ {
+					seg, err = s.PollSegment(segCtx)
+					if err == nil {
+						break
+					} else if err == context.Canceled || err == context.DeadlineExceeded {
```

Comment:
this could probably be combined with `err == nil || ...`

Reply:
Done